resources/function-controller/templates/secret.user-pass-secret.image-pull-secret.yaml

```diff
@@ -0,0 +1,37 @@
+{{- $username := include "tplValue" ( dict "value" .Values.secret.registryUserName "context" . ) -}}
+{{- $password := include "tplValue" ( dict "value" .Values.secret.registryPassword "context" . ) -}}
+apiVersion: v1
+kind: Secret
+type: kubernetes.io/basic-auth
+metadata:
+  name: {{ template "fullname" . }}-registry-credentials
+  namespace: {{ .Values.secret.namespace }}
+  labels:
+    {{- include "tplValue" ( dict "value" .Values.global.commonLabels "context" . ) | nindent 4 }}
+  annotations:
+    tekton.dev/docker-0: "http://{{- include "tplValue" ( dict "value" .Values.secret.registry "context" . )}}"
+stringData:
+  username: "{{ $username }}"
+  password: "{{ $password }}"
+---
+# when we migrate to copying secret from kyma-system to other namespaces please delete that file and use ONLY secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "fullname" . }}-registry-credentials
+  labels:
+  {{- include "tplValue" ( dict "value" .Values.global.commonLabels "context" . ) | nindent 4 }}
+stringData:
+  username: "{{ $username }}"
+  password: "{{ $password }}"
+---
+apiVersion: v1
+kind: Secret
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: "{{- printf "{\"auths\": {\"%s\": {\"auth\": \"%s\", \"email\":\"not@val.id\",\"username\":\"%s\",\"password\":\"%s\"}}}" (include "tplValue" ( dict "value" .Values.secret.externalRegistry "context" . )) (printf "%s:%s" $username $password | b64enc) $username $password | b64enc }}"
```

Comment:
Content annotation:
```
.dockerconfigjson: |
  {
     "auths": ...
      "password": {{ include "tplValue" ( dict "value" .Values.secret.externalRegistry "context" . ) }}
      etc...
  }
```

doesn't work? This is only question, because current implementation is very strange and not readable :D

Reply:
unfortunately this strings needs to be b64encoded, so it needs to be like that

besides it's taken from https://helm.sh/docs/howto/charts_tips_and_tricks/#creating-image-pull-secrets